dotnet/src/Experimental/Agents/Internal/Agent.cs

```diff
@@ -181,6 +196,12 @@ private AgentPluginImpl DefinePlugin()
         return new AgentPluginImpl(this, functionAsk);
     }
 
+    private IPromptTemplate DefinePromptTemplate(PromptTemplateConfig config)
+    {
+        var factory = new KernelPromptTemplateFactory(); // $$$ HANDLEBARS
```

Comment:
Should something happen with this comment?

Reply:
Yes...this seeped into my awareness and haunted me over the weekend...good call out. 

If I understand correctly, the appropriate template factory needs to be used depending on the template-type (defined withing the template).  This means its possible for the caller to use the wrong factory with a given template.

Ideally, I'd expect a single entry point that was able to fork on whether handlerbars or "kernel prompt"...zooming into thsi now for re-analysis.

Reply:
Updated